### PR TITLE
BUG: Add  type check for width parameter in str.pad method GH13598

### LINKED ISSUE #13598: Series.str.zfill() doesn't check type
#### Code Sample, a copy-pastable example if possible

```
In [5]: pd.Series(['a', 'b']).str.zfill('hi')
Out[5]: 
0   NaN
1   NaN
dtype: float64

```

#### Expected Output

A ``ValueError``.

#### output of ``pd.show_versions()``

```

In [3]: pd.show_versions()

INSTALLED VERSIONS
------------------
commit: a63bd12529ff309d957d714825b1753d0e02b7fa
python: 3.5.1.final.0
python-bits: 64
OS: Linux
OS-release: 4.5.0-2-amd64
machine: x86_64
processor: 
byteorder: little
LC_ALL: None
LANG: it_IT.utf8
LOCALE: it_IT.UTF-8

pandas: 0.18.1+174.ga63bd12
nose: 1.3.7
pip: 1.5.6
setuptools: 18.4
Cython: 0.23.4
numpy: 1.10.4
scipy: 0.16.0
statsmodels: 0.8.0.dev0+111ddc0
xarray: None
IPython: 5.0.0.dev
sphinx: 1.3.1
patsy: 0.3.0-dev
dateutil: 2.2
pytz: 2012c
blosc: None
bottleneck: 1.1.0dev
tables: 3.2.2
numexpr: 2.5
matplotlib: 1.5.1
openpyxl: None
xlrd: 0.9.4
xlwt: 1.1.2
xlsxwriter: 0.7.3
lxml: None
bs4: 4.4.0
html5lib: 0.999
httplib2: 0.9.1
apiclient: 1.5.0
sqlalchemy: 1.0.11
pymysql: None
psycopg2: None
jinja2: 2.8
boto: 2.38.0
pandas_datareader: 0.2.1

```

### PR DESCRIPTION
**Pull request by wcwagner**
- [x] closes #13598 
- [x] tests added / passed
- [x] passes `git diff upstream/master | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by sinhrks** [comment]
Thx! can u add tests all methods uses ``str_pad``, i.g. ``pad``, ``center``, ``ljust``, ``rjust`` and ``zfill``.

File: doc/source/whatsnew/v0.19.0.txt [hunk 1]
```diff
@@ -614,3 +614,4 @@ Bug Fixes
 - Bug in ``groupby`` with ``as_index=False`` returns all NaN's when grouping on multiple columns including a categorical one (:issue:`13204`)
 
 - Bug where ``pd.read_gbq()`` could throw ``ImportError: No module named discovery`` as a result of a naming conflict with another python package called apiclient  (:issue:`13454`)
+- Bug in ``pandas.Series.str.zfill``, no TypeErrors raised when  ``width`` was not of integer type (:issue:`13598`)
```
Comment (sinhrks) [inline doc/source/whatsnew/v0.19.0.txt]: nice to describe the results are all ``NaN``. Also,  ``` ``TypeError`` ```

**Comment by wcwagner** [comment]
I wrote all the tests, however it seemed extremely redundant doing so. I know that redundancy isn't necessarily a bad thing when testing, but is there a better way to implement them?

Thanks

File: pandas/tests/test_strings.py [hunk 2]
```diff
@@ -1603,6 +1603,18 @@ def test_pad_fillchar(self):
                                    "fillchar must be a character, not int"):
             result = values.str.pad(5, fillchar=5)
 
+    def test_pad_width(self):
```
Comment (jreback) [inline pandas/tests/test_strings.py]: you can just put them all in a single test method and iterate over the functions asserting that they raise

Comment (wcwagner) [inline pandas/tests/test_strings.py]: @jreback I'm relatively new to OOP in python, so could you let me know if there's anything I could do to improve my current implementation?
Thanks

File: pandas/core/strings.py [hunk 3]
```diff
@@ -914,6 +914,10 @@ def str_pad(arr, width, side='left', fillchar=' '):
     if len(fillchar) != 1:
         raise TypeError('fillchar must be a character, not str')
 
+    if not isinstance(width, compat.integer_types):
+        msg = 'width must be of integer type, not {0}'
```
Comment (jreback) [inline pandas/core/strings.py]: use is_integer

File: pandas/tests/test_strings.py [hunk 4]
```diff
@@ -1603,6 +1603,16 @@ def test_pad_fillchar(self):
...
+        values = Series(['1', '22', 'a', 'bb'])
+        string_methods = Series.str(values)
+
+        for f_name, f in Series.str.__dict__.items():
```
Comment (jreback) [inline pandas/tests/test_strings.py]: just iterate over the list and use getattr(s.str,f)

**Comment by jreback** [comment]
thanks @wcwagner

**Pull request closed without merging**